tests/spec/test_callback.py

```diff
@@ -0,0 +1,132 @@
+                    }
+                }
+            }
+        }
+        callback = Callback.from_dict(data)
+        assert callback is not None
+        assert "{$request.body#/callbackUrl}" in callback
+        assert callback["{$request.body#/callbackUrl}"] is not None
+
+    def test_callback_dict_access(self):
+        """Test dict-like access to callback expressions."""
+        data: dict[str, Any] = {
+            "expression1": {"post": {"responses": {"200": {}}}},
+            "expression2": {"get": {"responses": {"200": {}}}},
+        }
+        callback = Callback.from_dict(data)
+
+        assert "expression1" in callback
+        assert "expression2" in callback
+        assert "expression3" not in callback
+
+        assert callback["expression1"] is not None
+        assert callback["expression2"] is not None
+
+    def test_callback_iteration(self):
+        """Test iterating over callback expressions."""
+        data: dict[str, Any] = {
+            "expr1": {"post": {}},
+            "expr2": {"get": {}},
+        }
+        callback = Callback.from_dict(data)
+
+        keys = list(callback.keys())
+        assert "expr1" in keys
+        assert "expr2" in keys
+
+        values = list(callback.values())
+        assert len(values) == 2
+
+
+class TestCallbackParsing:
+    """Tests for parsing callbacks from OpenAPI specs."""
+
+    @pytest.fixture
+    def callback_spec_path(self) -> Path:
+        """Return path to callback example spec."""
+        return Path(__file__).parent.parent / "fixtures" / "callback_example.yaml"
+
+    def test_parse_spec_with_callbacks(self, callback_spec_path: Path):
+        """Test parsing a spec that contains callbacks."""
+        spec = parse_spec_from_file(callback_spec_path)
+        assert spec is not None
+        assert spec.version.major == 3
+        assert spec.version.minor == 0
+
+    def test_components_callbacks_parsed(self, callback_spec_path: Path):
+        """Test that callbacks in components are properly parsed."""
+        spec = parse_spec_from_file(callback_spec_path)
+
+        # Should have callbacks in components
+        assert len(spec.components.callbacks) > 0
+        assert "statusCallback" in spec.components.callbacks
+        assert "dataCallback" in spec.components.callbacks
+
+    def test_callback_structure(self, callback_spec_path: Path):
+        """Test that callback objects have correct structure."""
+        spec = parse_spec_from_file(callback_spec_path)
+
+        # Get statusCallback
+        status_callback = spec.components.callbacks["statusCallback"]
+        assert status_callback is not None
+        assert isinstance(status_callback, Callback)
+
+        # Check it has expressions
+        assert "{$request.body#/statusUrl}" in status_callback
+
+        # Check we can access the path item
+        path_item = status_callback["{$request.body#/statusUrl}"]
+        assert path_item is not None
+        assert "post" in path_item
```

Comment:
Probably need to change all these tests as a lot of the __dunder__ methods are arguably not relevant.